### PR TITLE
Update sql status to not report changes for unstaged tables that have changed but not visibly.

### PR DESCRIPTION
**Pull request by nicktobey**
Sometimes tables can change their hashes despite having no visible changes. The two ways I identified where this can happen are:

- The table has different column tags from the ancestor because it was originally a different table that got renamed.
- The table stores whether it's in a conflict and stores the hash of the other branch's table and the common ancestor table.

If these are the only changes to a table, we shouldn't report it as changed in `dolt status`.

These tables can still be staged and report as modified once staged.

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>f19e655</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>f19e655</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: integration-tests/bats/sql-status.bats [hunk 1]
```diff
@@ -150,5 +150,33 @@ teardown() {
...
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 0 ]
+
+    run dolt diff t1
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 0 ]
+
+    dolt add t1
+
+    dolt sql -r csv -q "select * from dolt_status;"
```
Comment (tbantle22) [inline integration-tests/bats/sql-status.bats:178]: Can remove this line?

**Review by tbantle22 — APPROVED** [review]
lgtm thanks for fixing

File: integration-tests/bats/sql-status.bats (hunk 1 above)
Comment (nicktobey) [reply integration-tests/bats/sql-status.bats]: Done

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>2dbd1f6</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>2dbd1f6</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**